packages/contracts-create2factory/src/utils/etherscanUtils.ts

```diff
@@ -0,0 +1,138 @@
+import { VerifyEtherscanParameters } from "../types/buildinfo.js";
+
+export async function etherscanVerifySourceCode({
+    apiUrl,
+    apiKey,
+    contractAddress,
+    sourceCode,
+    contractName,
+    compilerVersion,
+    constructorArguments,
+    evmVersion,
+}: VerifyEtherscanParameters) {
+    const params = {
+        apikey: apiKey,
+        module: "contract",
+        action: "verifysourcecode",
+        contractaddress: contractAddress,
+        sourceCode,
+        codeformat: "solidity-standard-json-input",
+        contractname: contractName,
+        compilerversion: compilerVersion,
+    } as any;
+
+    if (constructorArguments) params.constructorArguements = constructorArguments;
+    if (evmVersion) params.evmversion = evmVersion;
+
+    const parameters = new URLSearchParams(params);
+    const url = new URL(apiUrl);
+
+    const response = await fetch(url, {
+        method: "POST",
+        body: parameters,
+        headers: { "Content-Type": "application/x-www-form-urlencoded" },
+    });
+
+    return response;
+}
+
+export async function etherscanCheckVerifyStatus({
+    apiUrl,
+    apiKey,
+    guid,
+}: {
+    apiUrl: string;
+    apiKey: string;
+    guid: string;
+}) {
+    const params = {
+        apikey: apiKey,
+        module: "contract",
+        action: "checkverifystatus",
+        guid,
+    };
+    const parameters = new URLSearchParams(params);
+    const url = new URL(apiUrl);
+
+    const response = await fetch(url, {
+        method: "POST",
+        body: parameters,
+        headers: { "Content-Type": "application/x-www-form-urlencoded" },
+    });
+
+    const status = await response.json();
+    console.debug("Verification Status:", status);
+    return status;
+}
+
+export async function waitForEtherscanVerifyStatus({
+    apiUrl,
+    apiKey,
+    guid,
+    retries = 10,
+    interval = 5000,
+}: {
+    apiUrl: string;
+    apiKey: string;
+    guid: string;
+    retries?: number;
+    interval?: number;
+}): Promise<void> {
+    if (guid === "Contract source code already verified") {
```

Comment:
This seems out of the scope of this function as this is some edge case where you're passing a string that's not even a guid. This check should rather be done where it makes sense, specifically in `verifyContract` where you should check before calling `waitForEtherscanVerifyStatus`